### PR TITLE
Fix the build. Sinon can't stub methods that don't exist.

### LINKED ISSUE #6429: listenOnceV2 tests fail on saucelabs and/or integration
See https://travis-ci.org/ampproject/amphtml/builds/180272766 for more info.

The error is:
```
Chrome 45.0.2454 (Linux 0.0.0) amp-analytics.visibility listenOnceV2 "before each" hook for "should work for visible=true spec" FAILED
	TypeError: Cannot stub non-existent own property IntersectionObserver
	    at Object.stub (/tmp/f93200a6a1824e811f2aca38d9f5b072.browserify:66221:35 <- node_modules/sinon/lib/sinon/collection.js:93:0)
	    at Context.<anonymous> (/tmp/f93200a6a1824e811f2aca38d9f5b072.browserify:29602:15 <- /home/travis/build/ampproject/amphtml/extensions/amp-analytics/0.1/test/test-visibility-impl.js:501:14)
```

For this reason, the `listenOnceV2` `should work for visible=true spec` test has been disabled. Please enable once fixed.

### PR DESCRIPTION
**Pull request by aghassemi**
Build has been failing on old Chrome since window.IntersectionObserver does not exist and sinon can't stub methods that don't exist.

Fixes #6429

### CODE REVIEW

File: extensions/amp-analytics/0.1/test/test-visibility-impl.js [hunk 1]
```diff
@@ -498,6 +498,7 @@ describe('amp-analytics.visibility', () => {
       unobserveSpy = sandbox.stub();
       callbackSpy1 = sandbox.stub();
       callbackSpy2 = sandbox.stub();
+      ampdoc.win.IntersectionObserver = {};
```
Comment (jridgewell) [inline extensions/amp-analytics/0.1/test/test-visibility-impl.js]: This will permanently set `IntersectionObserver` on the test runner's window.

Comment (aghassemi) [inline extensions/amp-analytics/0.1/test/test-visibility-impl.js]: Good catch, skipping if no `ampdoc.win.IntersectionObserver`, can't find a way to stub non-existing methods with sinon.

File: extensions/amp-analytics/0.1/test/test-visibility-impl.js [hunk 2]
```diff
@@ -493,7 +493,10 @@ describe('amp-analytics.visibility', () => {
     let callbackSpy1;
     let callbackSpy2;
 
-    beforeEach(() => {
+    beforeEach(function() {
+      if (!ampdoc.win.IntersectionObserver) {
+        this.skip();
```
Comment (jridgewell) [inline extensions/amp-analytics/0.1/test/test-visibility-impl.js]: Use a [`TestConfig`](https://github.com/ampproject/amphtml/blob/master/test/_init_tests.js#L69), please.

Comment (jridgewell) [inline extensions/amp-analytics/0.1/test/test-visibility-impl.js]: Also, you'll need to return here, if we keep it as is.

**Pull request merged**